dataRetriever/interface.go

```diff
@@ -228,6 +228,12 @@ type HeadersPool interface {
 	GetNumHeaders(shardId uint32) int
 }
 
+// HeadersPoolSubscriber can subscribe for notifications when a new block header is added to the headers pool
+type HeadersPoolSubscriber interface {
```

Comment:
this should have been placed in consensus/interface.go

Reply:
moved to consensus/interface.go

Reply:
@SebastianMarian since it has only one dependency against the data package, we can create it in any package we want in order to reduce the dependencies as maximum. This is actually go idiomatic (create interfaces where used, as many time you need) and uncle's Bob recommendation in order to achieve higher decoupling and implement dependency inversion principle.

Reply:
ok, God bless uncles